src/Pate/MemCell.hs

```diff
@@ -75,121 +73,87 @@ instance PC.TestEquality (WI.SymExpr sym) => Eq (MemCell sym arch w) where
 instance PC.OrdF (WI.SymExpr sym) => Ord (MemCell sym arch w) where
   compare stamp1 stamp2  = PC.toOrdering $ PC.compareF stamp1 stamp2
 
--- | This is a collection of 'MemCell' that all represent memory regions of the same size.
---
--- Each 'MemCell' is associated with the predicate that says whether or not the
+-- | Each 'MemCell' is associated with the predicate that says whether or not the
 -- described memory is contained in the 'Pate.Equivalence.MemPred'.
-newtype MemCells sym arch w = MemCells (Map.Map (MemCell sym arch w) (WI.Pred sym))
+type MemCellPred sym arch = Map.Map (Some (MemCell sym arch)) (WI.Pred sym)
```

Comment:
Is this meant to be an abstraction maintained by this module? Could we make it a `newtype` to do so?

Reply:
Yes, although it is currently violated in a couple places that would be cumbersome to refactor.  I've wrapped it in a `newtype` an put up a ticket (#207) to clean up that boundary,